jupyter_client/session.py

```diff
@@ -849,10 +850,14 @@ def send(
         if stream and buffers and track and not copy:
             # only really track when we are doing zero-copy buffers
             tracker = stream.send_multipart(to_send, copy=False, track=True)
+            if isinstance(tracker, asyncio.Future):
+                tracker = tracker.result()
```

Comment:
This isn't going to work if the future's not done, is it? It should raise `InvalidStateError: Result is not set.`

Reply:
Yeah, I think I got lucky so far in that it was always done when I checked, looks like I'll have to use the shadow trick for `Session`.